Lucas

ijassiem opened the pull request.

ijassiem committed:
Added mod,divide,multiply

ijassiem committed:
Added functions and unit tests DGDT-14

ijassiem committed:
Create a pokemon class for instantiating pokemon characters.

This change consists of a base Pokemon class and a derived ElectricPokemon class.

These classes create pokemon characters. Methods are included for allowing the
pokemon character to speak, learn a new move, and print all its properties.
Unit tests for the classses are included.

Resolves: "DGDT-19"

ijassiem committed:
Add decorator-function for repeating methods in Pokemon class

This decorator @repeat provides the functionality of repeating the execution
of a method a random number of times, if the corresponding method is decorated.
The random number is generated ranging from 0-10. This decorator is used with
speak method in the Pokemon class. A unit test is included for ensuring that the
number of times the speak method is repeated does not exceed maximum number of
repeats which is 10.

ijassiem committed:
Create a python module to provide a sequence of Lucas numbers.

This module contains a generator function and a class object iterator to provide a sequence of Lucas numbers.
Unit tests for the above are included.

Resolves: "DGDT-33"

tyronevb reviewed (commented):
I noticed that this particular pull request doesn't have a summary message / description.

Did you intentionally/manually remove it when creating this pull request?

james-smith-za commented on cash_register.egg-info/top_level.txt:
```diff
@@ -0,0 +1,2 @@
+cash_register
+pokemon
```
Please set up your `.gitignore` file.

I'd suggest the following at least:
```
*.egg-info
__pycache__
.mypy_cache
```

ijassiem replied:
Noted.

james-smith-za commented on lucas/__init__.py:
```diff
@@ -0,0 +1 @@
+"""This is left blank."""
```
```suggestion
"""This file intentionally left blank."""
```

ijassiem replied:
Noted.

james-smith-za commented on lucas/lucas.py:
```diff
@@ -0,0 +1,83 @@
+
+        Returns
+        ------
+        obj
```
So `object` is one of the most basic types in Python. Just about anything is an `object`, so this doesn't actually tell you much. Since your function does `return self` it may be more useful to indicate explicitly that the return is of the same type as the object.
```suggestion
        MyLucas
```

ijassiem replied:
Noted.

james-smith-za commented on testing/test_lucas.py:
```diff
@@ -0,0 +1,34 @@
+"""Test module."""
+
+import lucas.lucas as p
```
Why did you import as `p`?

If it's to save typing, you could have done something like `from lucas.lucas import MyLucas`. To call the namespace just `p` makes the meaning less clear in your code below.

ijassiem replied:
Yes it is to save typing.

james-smith-za commented on testing/test_lucas.py:
```diff
@@ -0,0 +1,34 @@
+        """Test lucas class iterator."""
+        my_obj = p.MyLucas(1000)
+        cnt = 0
+        for val in my_obj:
+            print(val, "---", self.LUCAS_LIST[cnt])
+            assert val == self.LUCAS_LIST[cnt]
+            cnt += 1
```
A cool Python trick you could have used to save yourself having to increment `cnt` manually:

```suggestion
        for cnt, val in enumerate(my_obj):
            print(val, "---", self.LUCAS_LIST[cnt])
            assert val == self.LUCAS_LIST[cnt]
```

ijassiem replied:
Thanks, I forgot about enumerate.

james-smith-za commented on testing/test_lucas.py:
```diff
@@ -0,0 +1,34 @@
+        my_obj = p.MyLucas(1000)
+        cnt = 0
+        for val in my_obj:
+            print(val, "---", self.LUCAS_LIST[cnt])
```
While it's not bad or wrong to use a print statement in a unit test, just be aware that `pytest` doesn't display the output from your tests by default.

I got to see it by using the following command:
```
pytest -k lucas -rP
```

ijassiem replied:
I used `pytest -sv`.

james-smith-za reviewed (changes requested):
I'd suggest, before you do any more branching, merge your changes back into your `main` branch from all previous reviews.

If you want to see why this is an issue, chat to me during the consulting time today and I'll show you a few things you may not have been aware of.

ijassiem committed:
Implement changes relating to comments in pull request.

These changes pertains to the comments indicated in the pull request: "Lucas #5".

Editted docstring in __init__.py and lucas.py. Added enumerate() with for loop in test_lucas.py

Contributes to: DGDT-33
